desugarer.go

```diff
@@ -301,8 +335,33 @@ func desugar(astPtr *astNode, objLevel int) (err error) {
 	case *astIndex:
 		return unimplErr
 
+	case *astSlice:
+		stdslice := &astVar{id: "std.slice"}
+		var apply astNode
```

Comment:
I need *astNode (a pointer to astNode) and this is a different type from **astVar. And I didn't know I can do `var x type = y` :-). 

done